ambry-server/src/main/java/com/github/ambry/server/AmbryServerRequests.java

```diff
@@ -417,7 +423,62 @@ private AdminResponse handleBlobStoreControlRequest(DataInputStream requestStrea
    */
   private AdminResponse handleHealthCheckRequest(DataInputStream requestStream, AdminRequest adminRequest)
       throws StoreException, IOException {
-    return null;
+    boolean hostHealthy = true; //used to determine if the host is ever unhealthy
+
+    //Finds all replicas of every partition that have this server's hostName
+    List<PartitionId> partitionsWithHostName = new ArrayList<>();
+    StorageManager storageManager = (StorageManager)this.storeManager;
+    for(Map.Entry<PartitionId, DiskManager> entry : storageManager.getPartitionToDiskManager().entrySet()){
+      for(ReplicaId replica : entry.getKey().getReplicaIds()){
+        if (replica.getDataNodeId().getHostname() == currentNode.getHostname()) {
+          partitionsWithHostName.add(entry.getKey());
+          break;
+        }
+      }
+    }
+
+    /**
+     * Checks if the Host's BlobStore exists/initialized and is Leader/Standby ReplicaState for all Partitions
+     * this Host is apart of
+     */
+    ConcurrentHashMap<PartitionId, DiskManager> partitionToDiskManager = storageManager.getPartitionToDiskManager();
```

Comment:
doesn't seem like this is used anywhere.

Reply:
Removed